RFC/src/RFC-0130_Mining.md

```diff
@@ -1 +1,93 @@
+
+The solution to the PoW problem could be solved at the difficulty of either the Tari and/or Monero blockchain. If a solution has been found that meets the minimum difficulty requirements of the Monero blockchain, then the new Tari block should be updated by including the solved Monero block’s information (block header hash, Merkel tree branch, and hash of the coinbase transaction) into the PoW summary section of the Tari block header. If a solution was found that meet the Tari or Monero blockchain difficulty then the new Tari block can be added to the Tari blockchain. If the solution met the difficulty requirements of the Monero blockchain then the new Monero block can also be added to the Monero blockchain. If the PoW solution was sufficient to meet the difficult level of both the Tari and Monero blockchains then the individual blocks for each cryptocurrency can be added to their respective blockchains. The Tari block times is (hard fork) flexible and can be less than, equal or greater than the Monero block times.
+
+Solved and completed blocks should be propagated to the rest of the base layer network so that Nakamoto consensus can be performed on the new block and it can be added to the local blockchain copies of the full mining nodes.
+
+####  Primary functionality required by a Tari full-node miner
+- The Tari blockchain MUST have the ability to be merged mined with Monero using the latest released version of the Monero CryptoNight PoW algorithm.
+- The Tari full-node miner MUST maintain complete or pruned copies of the Tari and Monero blockchains.
+- It MUST be able to transmit and propagate information on the Tari base layer network using peer-to-peer communication using a gossip protocol. It SHOULD also propagate information and blocks on the Monero network.
+- It MUST have a mechanism to construct a new Tari and Monero block by selecting transactions from the different Tari and Monero mempools that need to be included in the different blocks.
+- It MUST have the ability to include the block header hash of the new Tari block into the coinbase section of a newly created Monero block to enable merged mining.
+- It MUST be able to include the Monero block header hash, Merkel tree branch and hash of the coinbase transaction of the Monero block into the PoW summary field of the new Tari block header. 
+- It MUST have the ability to perform a PoW algorithm on the newly created Monero block, that contains the Tari block information.
+- It MUST attempt to solve the PoW algorithm at the Tari and/or Monero difficulties. 
+- Valid Tari and/or Monero blocks SHOULD be propagated to other mining nodes on the Tari base layer network and/or the Monero network. 
+- The Tari full-node miner MUST reject and discard invalid Tari and/or Monero blocks.  
+- The Tari full-node miner MAY be implemented as a one click installer/miner with a user-friendly GUI.
```

Comment:
Suggest SHOULD instead of MAY. We should really aim for this.

Reply:
This requirement was removed.